### DISCUSSION TITLE
Sending binary data / Messagepack

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [ruby-amqp/bunny#654] [Q&A] [answered]
Hi,

I tried to send binary data (compressed msgpack data) via bunny, but it's not working and shows only "incompatible character encodings: UTF-8 and ASCII-8BIT".

Does the bunny gem support to send binary data?

Thanks for your help!

**Comment by h0jeZvgoxFepBQ2C** [comment, marked as answer]
Sorry, my fault, the error occured due to a wrong print statement on my side, msgpack is actually a string and can be sent without problems.

**Reply by michaelklishin** [reply]
Yes, neither RabbitMQ clients nor the broker care about the data you send in the payload. Compressed and/or binary formats are perfectly fine and have benefits compared to JSON.
